third_party/move/move-vm/runtime/src/move_vm.rs

```diff
@@ -44,11 +48,25 @@ impl MoveVM {
         natives: impl IntoIterator<Item = (AccountAddress, Identifier, Identifier, NativeFunction)>,
         vm_config: VMConfig,
     ) -> Self {
+        let natives = NativeFunctions::new(natives)
+            .unwrap_or_else(|e| panic!("Failed to create native functions: {}", e));
         Self {
             runtime: VMRuntime::new(natives, vm_config),
         }
     }
 
+    pub fn new_with_runtime_environment(runtime_environment: &RuntimeEnvironment) -> Self {
+        // Loader V2 does not store any natives, so we save the clone here.
```

Comment:
A bit confusing comment. Seems like runtime_envoronment.natives() should be empty here? Maybe debug_assert?

Reply:
This was confusing, fixed. Before, what I did is just avoided clone because V2 loader does not store natives (but they are stored in runtime environment). After addressing comments: removed `MoveVM::new` and `MoveVM::new_with_config` to ensure VM is always created based on the specified runtime environment (`MoveVM::new_with_runtime_environment`). Now there is less duplication, and the code is clear, and less clones in tests.